module/VuFind/src/VuFind/RecordDriver/MarcAdvancedTrait.php

```diff
@@ -903,4 +903,33 @@ public function getConsortialIDs()
     {
         return $this->getFieldArray('035', 'a', true);
     }
+
+    /**
+     * Return first ISMN found for this record, or false if no one fonund
+     *
+     * @return mixed
+     */
+    public function getCleanISMN()
+    {
+        $fields024 = $this->getMarcRecord()->getFields('024');
+        $ismn = null;
+        foreach ($fields024 as $field) {
+            if ($field->getIndicator(1) == 2) {
+                $ismn = $field->getSubfield('a')->getData();
+                break;
+            }
+        }
+        return $ismn ?? false;
+    }
+
+    /**
+     * Return first national bibliography number found, or false if not found
+     *
+     * @return mixed
+     */
+    public function getCleanNBN()
+    {
+        $nbns = $this->getFieldArray('015', ['a']);
```

Comment:
Do we need to account for subfield b in some fashion, in case multiple national bibliographies are represented in the index?

Reply:
There is no subfield b defined in MARC21: https://www.loc.gov/marc/bibliographic/bd015.html
In case somebody needs more NBNs, the 015 field is repetable. What we could change here is to return all NBNs, instead of first one.

Reply:
Sorry, I misspoke -- I meant `$2` rather than `$b` -- I'm guessing that the source of the number might be significant in terms of translating that number into an image.

Reply:
Sorry, I misspoke -- I meant `$2` rather than `$b` -- I'm guessing that the source of the number might be significant in terms of translating that number into an image.